snowflake/src/integration/java/org/apache/iceberg/snowflake/NamespaceTests.java

```diff
@@ -0,0 +1,268 @@
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public class NamespaceTests extends SnowTestBase {
+
+  @Test
+  public void testListNamespacesAtRootLevel() throws SQLException, InterruptedException {
+    List<Namespace> namespaces = snowflakeCatalog.listNamespaces(Namespace.empty());
+    int dbCount =
+        clientPool.run(
+            conn -> {
+              int databaseCount = 0;
+              ResultSet rs = conn.createStatement().executeQuery("show databases");
+              while (rs.next()) {
+                databaseCount++;
+              }
+              return databaseCount;
+            });
+    Assertions.assertThat(namespaces.stream().count()).isEqualTo(dbCount);
+  }
+
+  @Test
+  public void testListNamespacesAtDatabaseLevel() throws SQLException, InterruptedException {
+    String schema1 = "Schema_1";
+    String schema2 = "Schema_2";
+    String dbName = TestConfigurations.getInstance().getDatabase().toUpperCase();
+    try {
+      clientPool.run(conn -> conn.createStatement().execute("use " + dbName));
+      createOrReplaceSchema(schema1);
+      createOrReplaceSchema(schema2);
+      List<Namespace> namespaces = snowflakeCatalog.listNamespaces(Namespace.of(dbName));
+      Assertions.assertThat(namespaces)
+          .containsExactlyInAnyOrderElementsOf(
+              Arrays.asList(
+                  Namespace.of(dbName, schema1.toUpperCase()),
+                  Namespace.of(dbName, schema2.toUpperCase()),
+                  Namespace.of(dbName, "PUBLIC"),
+                  Namespace.of(dbName, "INFORMATION_SCHEMA")));
+    } finally {
+      dropSchemaIfExists(schema1);
+      dropSchemaIfExists(schema2);
+    }
+  }
+
+  @Test
+  public void testListNamespacesAtSchemaLevelIsNotAllowed()
+      throws SQLException, InterruptedException {
+    String schema = "Schema_1";
+    String dbName = TestConfigurations.getInstance().getDatabase().toUpperCase();
+    try {
+      clientPool.run(conn -> conn.createStatement().execute("use " + dbName));
+      createOrReplaceSchema(schema);
+      Assertions.assertThatThrownBy(
+              () -> {
+                snowflakeCatalog.listNamespaces(Namespace.of(dbName, schema));
+              })
+          .isInstanceOf(IllegalArgumentException.class)
+          .hasMessageContaining("listNamespaces must be at either ROOT or DATABASE level");
+    } finally {
+      dropSchemaIfExists(schema);
+    }
+  }
+
+  @Test
+  public void testLoadNonExistingRootLevelNamespace() {
+    String nonExistingDb = "IDontExist";
+    Assertions.assertThatThrownBy(
+            () -> {
+              snowflakeCatalog.loadNamespaceMetadata(Namespace.of(nonExistingDb));
+            })
+        .isInstanceOf(NoSuchNamespaceException.class);
```

Comment:
it is generally good pratice to assert for a specific error message, so would be good to have `.hasMessageContaining` here and all the other places below